exercises/concept/gotta-snatch-em-all/src/test/java/GottaSnatchEmAllTest.java

```diff
@@ -0,0 +1,197 @@
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.api.Test;
+
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+class GottaSnatchEmAllTest {
+
+    @Test
+    @Tag("task:1")
+    @DisplayName("newCollection returns an empty set when given an empty list")
+    void testNewCollectionEmptyList() {
+        assertThat(GottaSnatchEmAll.newCollection(List.of())).isEmpty();
+    }
+
+    @Test
+    @Tag("task:1")
+    @DisplayName("newCollection returns a set with one card when given a list with one card")
+    void testNewCollectionSingletonList() {
+        List<String> cards = List.of("Bleakachu");
+        Set<String> expected = Set.of("Bleakachu");
+        assertThat(GottaSnatchEmAll.newCollection(cards)).isEqualTo(expected);
+    }
+
+    @Test
+    @Tag("task:1")
+    @DisplayName("newCollection returns a set with one card when given a list with one repeated card")
+    void testNewCollectionListWithDuplicates() {
+        List<String> cards = List.of("Bleakachu", "Bleakachu");
+        Set<String> expected = Set.of("Bleakachu");
+        assertThat(GottaSnatchEmAll.newCollection(cards)).isEqualTo(expected);
+    }
+
+    @Test
+    @Tag("task:1")
+    @DisplayName("newCollection returns a set with two cards when given a list with two unique cards")
+    void testNewCollectionListWithoutDuplicates() {
+        List<String> cards = List.of("Bleakachu", "Newthree");
+        Set<String> expected = Set.of("Bleakachu", "Newthree");
+        assertThat(GottaSnatchEmAll.newCollection(cards)).isEqualTo(expected);
+    }
+
+    @Test
+    @Tag("task:2")
+    @DisplayName("addCard returns true when the collection did not yet contain the new card")
+    void testAddCardReturnsTrueWhenCardNotInCollection() {
+        Set<String> collection = new HashSet<>();
+        assertThat(GottaSnatchEmAll.addCard("Veevee", collection)).isTrue();
+    }
+
+    @Test
+    @Tag("task:2")
+    @DisplayName("addCard returns false when the collection already contains the new card")
+    void testAddCardReturnsFalseWhenCardAlreadyInCollection() {
+        Set<String> collection = new HashSet<>(Set.of("Veevee"));
+        assertThat(GottaSnatchEmAll.addCard("Veevee", collection)).isFalse();
+    }
+
+    @Test
+    @Tag("task:2")
+    @DisplayName("addCard adds the card to the collection when it is a new card")
+    void testAddCardShouldAddNewCardToCollection() {
+        Set<String> collection = new HashSet<>();
+        Set<String> expected = Set.of("Veevee");
+        GottaSnatchEmAll.addCard("Veevee", collection);
+        assertThat(collection).isEqualTo(expected);
+    }
+
+    @Test
+    @Tag("task:2")
+    @DisplayName("addCard adds the card to the collection when it is a new card")
```

Comment:
```suggestion
    @DisplayName("addCard doesn't add the card to the collection when it already contains the new card")
```

Noticed the display name was the same as the one before this one.